src/components/EmailNotifications/EmailNotificationsProvider.js

```diff
@@ -0,0 +1,142 @@
+import React, {
```

Comment:
Nitpick: we may want to rename this file to be `EmailNotificationsApiProvider` for clarity.